### PR TITLE
✨ Allow custom iframe sandbox attributes in amp-consent

### PR DESCRIPTION
**Pull request by 0x7f**
In #38183 @alanorozco and I discussed options how to allow amp-consent's consent-ui iframe to trigger navigation events in the top window. This pull request allows specifying additional sandbox attributes similar to how users can specify them for `amp-iframe` (see [amp-iframe v0.1](https://github.com/ampproject/amphtml/blob/main/extensions/amp-iframe/1.0/README.md#sandbox-optional-) and [amp-iframe v1.0](https://github.com/ampproject/amphtml/blob/main/extensions/amp-iframe/0.1/amp-iframe.md#sandbox-)).

Context: it is possible to load the consent ui in `amp-consent` from an external URL. The URL provided via `promptUISrc` is loaded inside an iFrame. The code inside that iFrame can communicate with the parent page via `postMessage` calls to e.g. signal ready or fullscreen events. The iFrame is instantiated with limited sandbox attributes (`allow-scripts`, `allow-popups` and optionally `allow-same-origin`). Since neither `allow-top-navigation` nor `allow-top-navigation-by-user-activation` are added, the iFrame is not allowed to trigger a navigation event in the top frame. There are cases though where triggering a navigation event from within that iFrame is useful (see #38183 for details).

This PR allows specifying additional sandbox attributes so users have full control over the auto-generated iframe element.

### CODE REVIEW

**Comment by 0x7f** [comment]
See #38249 for an alternative implementation.

**Comment by 0x7f** [comment]
@erwinmombay can you please add someone else as a reviewer instead?

**Comment by 0x7f** [comment]
I improved the patch by tweaking the wording and adding tests.

In addition to the original issue of the PR, I ran into another issue with the `amp-consent` extension: I open a window from within the prompt iframe. This window inherits the sandbox attribute of the iframe. As `allow-forms` restriction is not lifted, I can not submit forms in that window. To solve the original issue of the PR, and the new `allow-forms` issue, I would love to have my patch merged so I can specify additional restrictions in my AMP sites.

I thought about adding a dedicated config option per restriction. But I find this cumbersome to maintain. Every time some needs a new restriction to be removed from the sandbox, a new config variable has to be added. That's why I think providing the whole list of restrictions is the better way to go here.

**Comment by 0x7f** [comment]
@alanorozco the CI checks propose you as a reviewer. Can you please have a look or delegate the review?

**Review by newmuis — CHANGES_REQUESTED** [review]
Did this ever go through a design review?  I would expect that changing iframe sandbox would need to be reviewed for security and privacy, and is a major change that should follow the [process for significant changes](https://github.com/ampproject/amphtml/blob/main/docs/contributing-code.md#process-for-significant-changes).

**Comment by 0x7f** [comment]
Hi @newmuis, as mentioned in #38711, I am more than happy to proceed with the design review process. Would you be able to recommend someone who could assist me throughout this process?

To further clarify my rationale behind this pull request, let's take a simple consent layer containing a "privacy policy" link as an example. When using `target=_blank`, the link opens in a new tab, enabling the user to read the policy in its entirety. However, since the link originates from within an iframe with sandbox restrictions, the new tab inherits these restrictions, causing the site to seem nonfunctional to the user without a clear reason.

To resolve this issue, the `allow-popups-to-escape-sandbox` sandbox attribute was introduced, allowing website owners to ensure that links opened from within the iframe operate correctly.

Therefore, I believe it is most beneficial to offer website owners the option to choose which sandbox attributes they wish to enable.

**Comment by erwinmombay** [comment]
@0x7f could we narrow the scope of the PR to only allow for `allow-top-navigation` and `allow-top-navigation-by-user-activation` as those were the two values that security had allowed us to add. The current implementation currently is open to add anything correct?

**Comment by 0x7f** [comment]
> @0x7f could we narrow the scope of the PR to only allow for `allow-top-navigation` and `allow-top-navigation-by-user-activation` as those were the two values that security had allowed us to add. The current implementation currently is open to add anything correct?

Yes, the current implementation does allow any attribute. Let me change the code.

**Comment by 0x7f** [comment]
@erwinmombay I updated the code. I was not really sure about the allowed sandbox attributes. In slack you mentioned different ones than in your comment above. I now allow `allow-popups-to-escape-sandbox`, `allow-top-navigation` and `allow-top-navigation-by-user-activation`. Let me know if the current state is fine.

File: extensions/amp-consent/0.1/consent-ui.js
Comment (erwinmombay) [inline extensions/amp-consent/0.1/consent-ui.js:41]: talking to security, they do not want `allow-top-navigation`, the other 2 they have OK'd

**Comment by erwinmombay** [comment]
@0x7f thanks for the updates on the code. i was having security review this PR with https://github.com/ampproject/amphtml/pull/38711 which is why i mentioned the other values.

File: extensions/amp-consent/0.1/consent-ui.js
Comment (0x7f) [reply extensions/amp-consent/0.1/consent-ui.js:516]: Yes, it is about adding further values. I did not want to touch the logic related to the already existing values (`allow-scripts`, `allow-popups` and `allow-same-origin`) to prevent breaking currently working setups. Therefore, I decided to just allow adding further sandbox values to the pre-defined list.

Comment (erwinmombay) [reply extensions/amp-consent/0.1/consent-ui.js:515]: sounds good. will merge

**Comment by erwinmombay** [comment]
@0x7f looks like theres a flakey test failing, would you mind rebasing and pushing the code out again. will merge once PR is green. thank you for your patience in all of this

**Comment by 0x7f** [comment]
@erwinmombay CI is green now :)

**Comment by erwinmombay** [comment]
merge is blocked by a comment from @newmuis but this should have been addressed through the security review so i am force merging

**Comment by 0x7f** [comment]
@erwinmombay do you already have a plan when the next AMP release will be? Last one was two weeks ago. Time for a new one ;)

**Comment by erwinmombay** [comment]
@0x7f we should be cutting a new experimental/beta today and it would go live to production next week

**Comment by JacopoBassan** [comment]
@erwinmombay hi, "next week" is closing today, any news for the go live? Thanks.

**Pull request merged**
